src/sentry/integrations/github/client.py

```diff
@@ -112,6 +113,10 @@ def create_token(self):
             },
         )
 
+    @transaction_start("GitHubClientMixin")
+    def get_file_url(self, repo, path):
+        return self.get_cached(path="/repos/{}/contents/{}".format(repo, path))["html_url"]
```

Comment:
Do we need to get the whole file or can we do a `HEAD` request to check if it exists?

Reply:
@manuzope lol we were going back and forth on this (we being @MeredithAnya and @scefali) and at first I was doing a HEAD request but when we realized GH returns the formatted URL that saves us the trouble of constructing it ourselves.

Reply:
We have to construct the URL for Gitlab though so we could use a head request there if we wanted.

Reply:
Maybe I'm missing something -- aren't we sending them the path? So we should have it?

Reply:
We pass in the file path but not the full URL for GitHub, so like `/src/sentry/integrations/github/client.py` vs. `https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/github/client.py` - I suppose we'd still have to tack on the line number though.